x-pack/plugins/security_solution/server/lib/risk_engine/utils/create_ds.ts

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+// This file is a copy of x-pack/plugins/alerting/server/alerts_service/lib/create_concrete_write_index.ts
+// original function create index instead of datastream, and their have plan to use datastream in the future
+// so we probably should remove this file and use the original when datastream will be supported
+
+import type { IndicesSimulateIndexTemplateResponse } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import type { Logger, ElasticsearchClient } from '@kbn/core/server';
+import { get } from 'lodash';
+import { retryTransientEsErrors } from './retry_es';
+
+export interface IIndexPatternString {
+  template: string;
+  pattern: string;
+  alias: string;
+  name: string;
+  basePattern: string;
+  secondaryAlias?: string;
+}
+
+interface ConcreteIndexInfo {
+  index: string;
+  alias: string;
+  isWriteIndex: boolean;
+}
+
+interface UpdateIndexMappingsOpts {
+  logger: Logger;
+  esClient: ElasticsearchClient;
+  totalFieldsLimit: number;
+  concreteIndices: ConcreteIndexInfo[];
+}
+
+interface UpdateIndexOpts {
+  logger: Logger;
+  esClient: ElasticsearchClient;
+  totalFieldsLimit: number;
+  concreteIndexInfo: ConcreteIndexInfo;
+}
+
+const updateTotalFieldLimitSetting = async ({
+  logger,
+  esClient,
+  totalFieldsLimit,
+  concreteIndexInfo,
+}: UpdateIndexOpts) => {
+  const { index, alias } = concreteIndexInfo;
+  try {
+    await retryTransientEsErrors(
+      () =>
+        esClient.indices.putSettings({
+          index,
+          body: { 'index.mapping.total_fields.limit': totalFieldsLimit },
+        }),
+      { logger }
+    );
+    return;
+  } catch (err) {
+    logger.error(
+      `Failed to PUT index.mapping.total_fields.limit settings for alias ${alias}: ${err.message}`
+    );
+    throw err;
+  }
+};
+
+// This will update the mappings of backing indices but *not* the settings. This
+// is due to the fact settings can be classed as dynamic and static, and static
+// updates will fail on an index that isn't closed. New settings *will* be applied as part
+// of the ILM policy rollovers. More info: https://github.com/elastic/kibana/pull/113389#issuecomment-940152654
+const updateUnderlyingMapping = async ({
```

Comment:
Did you talk to @ymao1 about potentially exposing these helpers from the alerting plugin, or perhaps creating a shared package we can both use?

Reply:
The ideal plan will be to just reuse the creation of DS from the alerts plugin https://github.com/elastic/kibana/issues/154266